unit-tests/Test/Booster/Pattern/Bool.hs

```diff
@@ -0,0 +1,168 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+{- |
+Copyright   : (c) Runtime Verification, 2022
+License     : BSD-3-Clause
+-}
+module Test.Booster.Pattern.Bool (
+    test_patternSynonyms,
+    test_conjunction_splitters,
+) where
+
+import Booster.Pattern.Base
+import Booster.Pattern.Bool
+import Booster.Syntax.ParsedKore.Internalise (symb)
+import Data.ByteString (ByteString)
+import Test.Booster.Fixture
+import Test.Tasty
+import Test.Tasty.HUnit
+
+--------------------------------------------------------------------------------
+test_patternSynonyms :: TestTree
+test_patternSynonyms =
+    testGroup
+        "Pattern synonyms for predicate symbols should match parsed symbol applications"
+        [ testPatternAndBool
+        , testPatternNotBool
+        , testPatternEqualsBool
+        , testPatternEqualsInt
+        , testPatternNEqualsInt
+        , testPatternEqualsK
+        , testPatternNEqualsK
+        , testPatternSetIn
```

Comment:
:+1: Very good, covers all hard-wired patterns.
What we could do in addition to this unit test is an integration test which _compiles_ a simple K definition (with `INT` and `BOOL` in scope) that contains specific target terms to match with the hard-wired patterns. That way we would notice changes in the compiler or the `domains.md` prelude.

Reply:
Good idea, I will do that!

Reply:
I've moved these tests to `test/predicates-integration`.